### DISCUSSION TITLE
How can I keep my API token secret in my pages/api.js file?

### DISCUSSION THREAD

**Discussion by zackshapiro** [vercel/vercel#4461] [Help]
I have a pretty simple setup where I have a `pages/api.js` file that requests some cards from Trello:

```
export default async (req, res) => {
  return await fetch(`https://api.trello.com/1/lists/abc123/cards?key=${process.env.key}&token=${process.env.token}`)
```

However, when I build and deploy my app and click into the .js files when I view source and search for Trello, I can see my API key and token right there. 

The docs are a bit confusing and all of the approaches I've tried don't seem to work (.env file, next.config.js file, and setting them in `https://vercel.com/{username}/{projectName}/settings`

A bit confused about what to do here to ensure the security of my `token` yet still be able to do my request and get the data my app requires.

Thanks!

**Comment by mcsdevv** [comment]
Hey @zackshapiro, can you tell us what method you've used to do this?
